tests/integration/framework/docker_runner.py

```diff
@@ -8,49 +8,93 @@
 import sys
 
 class DockerRunner:
-    """Run docker containers for testing"""
+    """Run a docker container in background to execute testing commands"""
 
     def __init__(self, platform, image, verbose):
         """Sets platform and image for all tests ran with this instance"""
         self.platform = platform
         self.image = image
         self.verbose = verbose
+        self.container_id = None
 
-    def construct_docker_command(self, envs, args):
+        # Launch the container
+        self._start()
+
+    def execute(self, envs, args):
+        """Launch `docker exec` commands inside the background container"""
+        command = self._construct_command("exec", envs, args)
+        return self._shell(command)
+
+    def run(self, envs, args):
+        """
+        Launch `docker run` commands to create a new container for each command
+        """
+        command = self._construct_command("run", envs, args)
+        return self._shell(command)
+
+    def __del__(self):
+        """Remove test container"""
+        stop_command = ["docker", "rm", "-f", self.container_id]
+        self._shell(stop_command, silent=True)
+
+    def _start(self):
         """
-        Construct a docker command with env and args
+        Launch a docker container. Done in 2 step using create + start.
+
+        Keep the container running with a shell open thanks to `--interactive`,
+        having stdin open keep the process running.
```

Comment:
Why is this better than `docker run -d <params> sleep infinity`?

Reply:
It's a way to do that. I agree `sleep infinity` would simplify the process.